src/quick-lint-js/fe/diagnostic-types.h

```diff
@@ -2582,6 +2582,14 @@
       MESSAGE(QLJS_TRANSLATABLE("using '{0}' against a regular "                \
                                 "expression literal always returns '{1}'"),     \
               equals_operator, comparison_result))                              \
+                                                                                \
+  QLJS_DIAG_TYPE(                                                               \
+      diag_this_parameter_not_allowed_in_typescript_constructor, "E0348",       \
+      diagnostic_severity::error, { source_code_span this_keyword; },           \
+      MESSAGE(                                                                  \
+          QLJS_TRANSLATABLE("in TypeScript, 'this' keyword is not allowed "     \
+                            "in constructors"),                                 \
```

Comment:
Nit: I think saying "in TypeScript" is confusing. Mentioning TypeScript implies that 'this' is allowed in JavaScript, but it is not allowed in JavaScript either. I think a better message is: "'this' keyword is not allowed in constructors"

Nit: Saying "'this' keyword is not allowed in constructors" is misleading. 'this' *is* allowed in constructors, but it is not allowed as a constructor *parameter*. I think a better message is: "'this' parameters are now allowed in constructors"